ports/espressif/bindings/espnow/ESPNow.c

```diff
@@ -0,0 +1,943 @@
+    mp_obj_list_t *list = MP_OBJ_TO_PTR(item->value);
+    list->items[0] = MP_OBJ_NEW_SMALL_INT(rssi);
+    list->items[1] = mp_obj_new_int(time_ms);
+    return item;
+}
+#endif // MICROPY_ESPNOW_RSSI
+
+// ### Handling espnow packets in the recv buffer
+//
+
+// ### Send and Receive ESP_Now data
+//
+
+// Return C pointer to byte memory string/bytes/bytearray in obj.
+// Raise ValueError if the length does not match expected len.
+static uint8_t *_get_bytes_len_rw(mp_obj_t obj, size_t len, mp_uint_t rw) {
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(obj, &bufinfo, rw);
+    if (bufinfo.len != len) {
+        mp_raise_ValueError(
+            MP_ERROR_TEXT("ESPNow: bytes or bytearray wrong length"));
+    }
+    return (uint8_t *)bufinfo.buf;
+}
+
+static uint8_t *_get_bytes_len(mp_obj_t obj, size_t len) {
+    return _get_bytes_len_rw(obj, len, MP_BUFFER_READ);
+}
+
+static uint8_t *_get_bytes_len_w(mp_obj_t obj, size_t len) {
+    return _get_bytes_len_rw(obj, len, MP_BUFFER_WRITE);
+}
+
+// Return C pointer to the MAC address.
+// Raise ValueError if mac_addr is wrong type or is not 6 bytes long.
+static const uint8_t *_get_peer(mp_obj_t mac_addr) {
+    return mp_obj_is_true(mac_addr)
+        ? _get_bytes_len(mac_addr, ESP_NOW_ETH_ALEN) : NULL;
+}
+
+// Copy data from the ring buffer - wait if buffer is empty up to timeout_ms
+static int ringbuf_read_wait(ringbuf_t *r, void *data, size_t len, int timeout_ms) {
+    int64_t end = mp_hal_ticks_ms() + timeout_ms;
+    int status = 0;
+    while (
+        ((status = ringbuf_read(r, data, len)) == 0) &&
+        (end - (int64_t)mp_hal_ticks_ms()) >= 0) {
+        RUN_BACKGROUND_TASKS;
+    }
+    return status;
+}
+
+// ESPNow.recvinto(buffers[, timeout_ms]):
+// Waits for an espnow message and copies the peer_addr and message into
+// the buffers list.
+// Arguments:
+//      buffers: (Optional) list of bytearrays to store return values.
+//      timeout_ms: (Optional) timeout in milliseconds (or None).
+// Buffers should be a list: [bytearray(6), bytearray(250)]
+// If buffers is 4 elements long, the rssi and timestamp values will be
+// loaded into the 3rd and 4th elements.
+// Default timeout is set with ESPNow.config(timeout=milliseconds).
+// Return (None, None) on timeout.
+STATIC mp_obj_t espnow_recvinto(size_t n_args, const mp_obj_t *args) {
+    esp_espnow_obj_t *self = _get_singleton_initialised();
+
+    size_t timeout_ms = ((n_args > 2 && args[2] != mp_const_none)
+            ? (size_t)mp_obj_get_int(args[2]) : self->recv_timeout_ms);
+
+    mp_obj_list_t *list = MP_OBJ_TO_PTR(args[1]);
+    if (!mp_obj_is_type(list, &mp_type_list) || list->len < 2) {
+        mp_raise_ValueError(MP_ERROR_TEXT("ESPNow.recvinto(): Invalid argument"));
+    }
+    mp_obj_array_t *msg = MP_OBJ_TO_PTR(list->items[1]);
+    if (mp_obj_is_type(msg, &mp_type_bytearray)) {
+        msg->len += msg->free;   // Make all the space in msg array available
+        msg->free = 0;
+    }
```

Comment:
use `mp_arg_parse_all()` style instead of checking individual args with indices.